botocore/handlers.py

```diff
@@ -432,6 +432,24 @@ def add_glacier_checksums(params, **kwargs):
     body.seek(starting_position)
 
 
+def document_glacier_tree_hash_checksum():
+    doc = '''
+        This is a required field.
+        You can use botocore.util.calculate_tree_hash() to calculate it
+        like this::
+
+            from botocore.utils import calculate_tree_hash
+
+            ...
+
+            client.complete_multipart_upload(
+                checksum=calculate_tree_hash(open('your_file.txt', 'rb')),
```

Comment:
FWIW, I don't think this is typically how a tree is calculated because you have to scan the file twice, which may be a fairly large file.

It's not technically wrong, but I typically see code that keeps the individual tree hashes as you're uploading parts and then aggregate the individual tree hashes all into a final tree hash (this is what boto does in its multipart archive uploader).

Might be worth pointing that out.